components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/core/OAuthParRequestWrapper.java

```diff
@@ -51,10 +56,22 @@ public OAuthParRequestWrapper(HttpServletRequest request, Map<String, String> pa
     @Override
     public String getParameter(String name) {
 
+        // Remove param request_uri to avoid conflicting with OIDC requests passed by reference.
+        if (StringUtils.equals(name, OAuthConstants.OAuth20Params.REQUEST_URI)) {
+            return null;
+        }
         if (params.containsKey(name)) {
             return params.get(name);
         }
-
         return super.getParameter(name);
     }
+
+    @Override
+    public Map<String, String[]> getParameterMap() {
+
+        Map<String, String[]> parameterMap = new HashMap<>(super.getParameterMap());
+        params.forEach((key, value) -> parameterMap.put(key, new String[]{value}));
+        parameterMap.remove(OAuthConstants.OAuth20Params.REQUEST_URI);
```

Comment:
Rather than removing this afterwards, we can include this as a condition in the above.